presto-main/src/main/java/io/prestosql/execution/resourcegroups/InternalResourceGroup.java

```diff
@@ -327,7 +324,7 @@ public void setSoftMemoryLimit(DataSize limit)
             boolean oldCanRun = canRunMore();
             this.softMemoryLimitBytes = limit.toBytes();
             if (canRunMore() != oldCanRun) {
-                updateEligibility();
+                updateEligibilityOnPathToRoot();
```

Comment:
@raghavsethi I completely removed the downward method (updateEligibilityInSubtree) after looking at the benchmark results. But I still kept the name `updateEligibilityOnPathToRoot` for the `updateEligibility` method,(without any changes to its implementation) since the new name is more descriptive. So this is just a name change overall.

`InternalResourceGroup#run` method checks the limits by invoking `canRunMore` on all the nodes from leaf to root. So I think we're okay here just updating eligibility up the path to root, and not worry about the nodes below it. Are you wondering if the current implementation should be changed?